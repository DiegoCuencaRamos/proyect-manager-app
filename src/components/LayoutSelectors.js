import React from 'react'

const LayoutSelectors = ({ handleLayoutChange }) => {
    
    const onAddClass = (e) => {
        const target = e.target
        const layoutSelectorItems = document.querySelectorAll('.layout-selector__item')
        const selectedClass = 'layout-selector__item--selected'
        
        layoutSelectorItems.forEach(item => {
            if(
                item === target 
                && !item.classList.contains(selectedClass)
            ) {
                item.classList.add(selectedClass)
            } else if (
                item === target 
                && item.classList.contains(selectedClass)
            ) {
                // Do nothing
            } else {
                item.classList.remove(selectedClass)
            }
        })
    }

    const onLayoutChange = e => {
        const layoutData = e.target.dataset.layout
        handleLayoutChange(layoutData)
    }

    return (
        <section className="layout-selector">
            <div className="container">
                <div className="layout-selector__wrapper" onClick={onAddClass}>
                    <div 
                        id="selectorItem1"
                        className="layout-selector__item" 
                        onClick={onLayoutChange}
                        data-layout="list"
                    >
                        <i className="fas fa-list-ul"></i>
                        <span>List</span>
                    </div>
                    <div 
                        id="selectorItem2"
                        className="layout-selector__item" 
                        onClick={onLayoutChange}
                        data-layout="calendar"
                    >
                        <i className="fas fa-calendar-alt"></i>
                        <span>Calendar</span>
                    </div>
                    <div 
                        id="selectorItem3"
                        className="layout-selector__item" 
                        onClick={onLayoutChange}
                        data-layout="kanban"
                    >
                        <i className="fas fa-columns"></i>
                        <span>Kanban</span>
                    </div>                       
                </div>
            </div>
        </section>
    )
}

export default LayoutSelectors