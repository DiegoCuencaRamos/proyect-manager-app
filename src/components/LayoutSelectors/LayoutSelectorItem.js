import React from 'react'

const LayoutSelectorItem = ({ handleLayoutChange, dataLayout, icon }) => {
    // 1. Event
    const onLayoutChange = e => {
        const layoutData = e.target.dataset.layout
        handleLayoutChange(layoutData)
    }
    
    // 2. Render
    return (
        <div 
            className="layout-selector__item" 
            onClick={onLayoutChange}
            data-layout={dataLayout}
        >
            <i className={icon}></i>
            <span>{dataLayout}</span>
        </div>
    )
}

export default LayoutSelectorItem