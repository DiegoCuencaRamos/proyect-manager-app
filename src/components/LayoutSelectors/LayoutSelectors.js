import React from 'react'
import LayoutSelectorItem from './LayoutSelectorItem'

const LayoutSelectors = ({ handleLayoutChange }) => {
    // 1. Event
    const onAddClass = (e) => {
        const target = e.target
        const layoutSelectorItems = document.querySelectorAll('.layout-selector__item')
        const selectedClass = 'layout-selector__item--selected'
        
        layoutSelectorItems.forEach(item => {
            if(
                item === target 
                && !item.classList.contains(selectedClass)
            ) {
                item.classList.add(selectedClass)
            } else if (
                item === target 
                && item.classList.contains(selectedClass)
            ) {
                // Do nothing
            } else {
                item.classList.remove(selectedClass)
            }
        })
    }
    
    //2. Render
    return (
        <section className="layout-selector">
            <div className="container">
                <div className="layout-selector__wrapper" onClick={onAddClass}>
                    <LayoutSelectorItem 
                        handleLayoutChange={handleLayoutChange}
                        dataLayout="list"
                        icon="fas fa-list-ul"
                    />
                    <LayoutSelectorItem 
                        handleLayoutChange={handleLayoutChange}
                        dataLayout="calendar"
                        icon="fas fa-calendar-alt"
                    />
                    <LayoutSelectorItem 
                        handleLayoutChange={handleLayoutChange}
                        dataLayout="kanban"
                        icon="fas fa-columns"
                    />                      
                </div>
            </div>
        </section>
    )
}

export default LayoutSelectors