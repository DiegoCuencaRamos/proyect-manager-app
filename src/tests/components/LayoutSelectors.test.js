import React from 'react'
import { shallow } from 'enzyme'
import LayoutSelectors from '../../components/LayoutSelectors'

let handleLayoutChange, wrapper

beforeEach(() => {
    handleLayoutChange = jest.fn()
    wrapper = shallow(<LayoutSelectors handleLayoutChange={handleLayoutChange} />)
})

test('Should render LayoutSelectors correctly', () => {
    expect(wrapper).toMatchSnapshot()
})

test('Should call handleLayoutChange on layout selector clicked', () => {
    const dataLayout = 'list'
    const layoutSelectorItems = wrapper.find('.layout-selector__wrapper').children()

    for(let i = 0; i < layoutSelectorItems.length; i++) {
        layoutSelectorItems.at(i).simulate('click', {
            target: { 
                dataset: { 
                    layout: dataLayout
                } 
            }
        })
        expect(handleLayoutChange).toHaveBeenCalledWith(dataLayout)
    }
})

// For the moment I wont test style event changes