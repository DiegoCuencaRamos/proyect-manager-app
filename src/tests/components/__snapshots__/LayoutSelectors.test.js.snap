// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LayoutSelectors correctly 1`] = `
<section
  className="layout-selector"
>
  <div
    className="container"
  >
    <div
      className="layout-selector__wrapper"
      onClick={[Function]}
    >
      <div
        className="layout-selector__item"
        data-layout="list"
        id="selectorItem1"
        onClick={[Function]}
      >
        <i
          className="fas fa-list-ul"
        />
        <span>
          List
        </span>
      </div>
      <div
        className="layout-selector__item"
        data-layout="calendar"
        id="selectorItem2"
        onClick={[Function]}
      >
        <i
          className="fas fa-calendar-alt"
        />
        <span>
          Calendar
        </span>
      </div>
      <div
        className="layout-selector__item"
        data-layout="kanban"
        id="selectorItem3"
        onClick={[Function]}
      >
        <i
          className="fas fa-columns"
        />
        <span>
          Kanban
        </span>
      </div>
    </div>
  </div>
</section>
`;
